test/unittest/tests/block/test_extent_list_insert_block.cpp

```diff
@@ -0,0 +1,383 @@
+            {
+              off_size(4096 - 64, 3 * 4096 + 64), // [4,032, 16,383], First'''.
+              off_size(5 * 4096, 4096),           // [20,480, 24,575] Third.
+            }},
+          {off_size(
+             6 * 4096, 64), // [20,480, 12,287] Just above First''', Merge First''' and Third.
+            {
+              off_size(4096 - 64, 3 * 4096 + 64), // [4,032, 16,383], First'''.
+              off_size(5 * 4096, 4096 + 64),      // [20,480, 24,639] Third'.
+            }},
+        };
+
+        /* Setup. */
+        WT_EXTLIST extlist = {};
+        extlist.name = const_cast<char *>("__block_merge");
+
+        WT_BLOCK block = {};
+        block.name = "__block_merge";
+        block.allocsize = 1024;
+        block.size = 4096; // Description information.
+
+        /* Test. */
+        /* Insert/merge extents and verify. */
+        int idx = 0;
+        for (const off_size_expected &test : test_list) {
+            /* Call. */
+            REQUIRE(__ut_block_merge(session, &block, &extlist, test.test_off_size.off,
+                      test.test_off_size.size) == 0);
+            INFO("After " << idx << ". Insert/merge: " << &test.test_off_size);
+
+            utils::extlist_print_off(extlist);
+
+            /* Verify. */
+            verify_off_extent_list(extlist, test.expected_list, true);
+            ++idx;
+        }
+
+        /* Cleanup. */
+        extlist_free(session, extlist);
+    }
+}
+
+/*!
+ * A test (off) and the expected value (expected_list) for operations that need an off to modify a
+ * WT_EXTLIST.
+ */
+struct off_expected {
+    wt_off_t off;
+    std::vector<off_size> expected_list;
+};
+
+TEST_CASE("Extent Lists: block_off_remove", "[extent_list]")
+{
+    /* Build Mock session, this will automatically create a mock connection. */
+    std::shared_ptr<MockSession> mock_session = MockSession::buildTestMockSession();
+    WT_SESSION_IMPL *session = mock_session->getWtSessionImpl();
+
+    std::vector<WT_EXT **> stack(WT_SKIP_MAXDEPTH, nullptr);
+
+    SECTION("remove multiple extents and verify all extents after each remove")
+    {
+        BREAK;
+        /* Extents to insert to setup for __ut_block_remove. */
+        std::vector<off_size> insert_list{
+          off_size(3 * 4096, 4096), // Second [12,288, 16,383].
+          off_size(4096, 4096),     // First [4,096, 8,191].
+          off_size(5 * 4096, 4096), // Third [20,480, 24,575].
+        };
+
+        /* Tests and expected values. */
+        std::vector<off_expected> test_list{
+          {3 * 4096, // [12,288, 16,383] Second.
+            {
+              off_size(4096, 4096),     // [4,096, 8,191] First.
+              off_size(5 * 4096, 4096), // [20,480, 24,575] Third.
+            }},
+          {4096, // [4,096, 8,191] First.
+            {
+              off_size(5 * 4096, 4096), // [20,480, 24,575] Third.
+            }},
+          {5 * 4096, // [20,480, 24,575] Third.
+            {}},
```

Comment:
Since there is an explicit test that the extent list is empty after performing all of the test cases, it might be worth a comment here (or when defining `test_list`, above, stating this assumption.

I.e., if somebody decides in the future to add/change the tests here, they might not realize the tests should leave the extent list empty.

Reply:
There is. nothing special about ending with an empty extent list. I just did that.

Every test has an expected result which depends upon both the test itself and every test that came before it. If any test is changed or added all the expected results after it must change, unless the added test is a NOP or the changed test has the same result as before.  That is implicit in testing the result after every test.

Reply:
Ah. I missed that. Ignore my suggestion then.